src/components/Resources/ResourcePage.js

```diff
@@ -3,46 +3,110 @@ import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
 import axios from 'axios';
 import PersonalMenu from '../PersonalMenu';
-import { Grid, Breadcrumbs, Typography } from '@material-ui/core';
+import { Grid, Breadcrumbs, Typography, Chip } from '@material-ui/core';
 import NavigateNextIcon from '@material-ui/icons/NavigateNext';
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles({
+  subtitle: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+});
 
 function ResourcePage({ matchProps }) {
   const [resource, setResource] = useState({});
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     axios
       .get('/api/v1/resources/' + matchProps.match.params.guid)
       .then(function(response) {
         // handle success
         setResource(response.data);
+        setLoading(false);
       })
       .catch(function(error) {
         // handle error
         console.log(error);
       });
-  }, [matchProps.match.params.id]);
+  }, [matchProps.match.params.guid]);
+
+  const classes = useStyles();
+
+  const {
+    title,
+    url,
+    author,
+    tags,
+    media_type: mediaType,
+    description,
+    user,
+    date_published: datePublished,
+    modified,
+    paid,
+  } = resource;
 
   return (
-    <>
-      <Grid container spacing={1}>
-        <Grid item lg={3}>
-          <PersonalMenu />
-        </Grid>
-        <Grid item lg={9}>
-          <Breadcrumbs
-            separator={<NavigateNextIcon fontSize="small" />}
-            aria-label="breadcrumb"
-          >
-            <Link color="inherit" to="/resources">
-              Resources
-            </Link>
-            <Typography color="textPrimary">{resource.title}</Typography>
-          </Breadcrumbs>
-          <h2>{resource.title}</h2>
-          <pre>{JSON.stringify(resource, 0, 2)}</pre>
-        </Grid>
+    <Grid container spacing={1}>
+      <Grid item lg={3}>
+        <PersonalMenu />
+      </Grid>
+      <Grid item lg={9}>
+        <Breadcrumbs
+          separator={<NavigateNextIcon fontSize="small" />}
+          aria-label="breadcrumb"
+        >
+          <Link color="inherit" to="/resources">
+            Resources
+          </Link>
+          <Typography color="textPrimary">{title}</Typography>
+        </Breadcrumbs>
+        {loading ? (
```

Comment:
This is fine, but may I suggest refactoring this out of the ResourcePage and turn into its own component? Some benefits: 

Turns the ResourcePage into a Higher Order Component where it's only responsible for data fetching and managing state (loading or not loading). It would simplify the responsibilities and tests.

Then the Resource component itself can be more stateless and only cares about rendering props. Maintainability would be easier as well - if we add or extend features, we know exactly where to do it.

Reply:
So we would be importing this component into the page itself?

Reply:
I might need some help with this, I can't remember what @lpatmo or you would prefer here--do we have a directory for custom components within the app?